### ISSUE TITLE
Breaking change in 3.0.3 - Unable to create deployment

### ISSUE THREAD

**Issue by rory-hughes-cko** [OctopusDeploy/create-release-action#373]
**Describe the bug**

Getting the below error when trying to create a deployment with `3.0.3`. Works correctly with `3.0.2`.

We're using Octopus 2022.3, so this seems like a bug. Regardless, I'd expect a patch version bump to not cause issues with users who were using an older version of Octopus with `3.0.2`

From looking at the PR for `3.0.3`, it seems a like a breaking might have been introduced by changing the API client from `2.1.3` to `3.0.1`

```
Run OctopusDeploy/create-release-action@v3
  with:
    api_key: ***
    server: ***
    project: My Project
    release_number: 1.2.3
    release_notes: my release notes
    ignore_existing: false
  env:
    APP_VERSION: 1.2.3
    APP_NAME: my-app
    OCTOPUS_SPACE: Default
    AWS_DEFAULT_REGION: eu-west-1
    AWS_REGION: eu-west-1
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
🐙 Creating a release in Octopus Deploy...
Error: The Octopus instance does not support the Capabilities API, you will need to upgrade it at least 2022.[3](https://github.com/cko-prism/checkout-prism-api/actions/runs/4053939449/jobs/6977100040#step:5:3) to get access to the feature you are trying to use.
Error: Error: The Octopus instance does not support the Capabilities API, you will need to upgrade it at least 2022.3 to get access to the feature you are trying to use.
```

**To Reproduce**
Create a deployment using `3.0.2` - works
Create a deployment using `3.0.3` - Fails with error message above

**Expected behavior**
Deployment created

**Screenshots**
![image](https://user-images.githubusercontent.com/57362462/216600446-aa6a03dd-8de4-4631-bc6d-1de9c8fe2a04.png)

**Desktop (please complete the following information):**
 - OS: ubunutu (via GitHub Actions)

**Smartphone (please complete the following information):**
 - n/a
 
**Additional context**

Usage of action as a GHA step
```
      - name: Create a Release
        uses: OctopusDeploy/create-release-action@v3
        with:
          api_key: <secret key>
          server:  <server url>
          project: <project>
          release_number: <release number>
          release_notes: <release notes>
```

**Comment by slewis74** [comment]
We updated some checks in the API client, to try to provide better messaging when the Octopus server version doesn't have the Executions API. Unfortunately the API we were using to do that check had an issue that we weren't aware of. We've updated the client to do the checks a different way and a new version of this action, plus the deployment and runbook run actions, is being shipped.
Apologies for the inconvenience this has caused.
